### PR TITLE
Fix AttributeError: 'NoneType' object has no attribute 'finalize'  on TPU

### PR DESCRIPTION
**Pull request by Chizuchizu**
Fix AttributeError: 'NoneType' object has no attribute 'finalize'

### Issue

```python
import pytorch_lightning as pl

my_model = MyLightningModule()
trainer = pl.Trainer(tpu_cores=8, logger=None)
trainer.fit(my_model)
```
If you run the code written above, you will get the error below.

```bash
/opt/conda/lib/python3.7/site-packages/pytorch_lightning/plugins/training_type/tpu_spawn.py in _close_logger(self, trainer)
    236     def _close_logger(self, trainer) -> None:
    237         if hasattr(trainer, "logger"):
--> 238             trainer.logger.finalize("success")
    239 
    240     @property

AttributeError: 'NoneType' object has no attribute 'finalize'

```

This bug is caused by not being able to handle the case where the attribute `logger` exists and `logger.finalize` does not.

### CHANGE

`pytorch_lightning/plugins/training_type/tpu_spawn.py`   the 249th line

BEFORE
```python
def _close_logger(self, trainer) -> None:
    if hasattr(trainer, "logger"):
        trainer.logger.finalize("success")
```

AFTER
```python
def _close_logger(self, trainer) -> None:
    if hasattr(trainer, "logger.finalize"):
        trainer.logger.finalize("success")
```

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

This is the first pull request to OSS in my life.  So,  of course, yes:)

### CODE REVIEW

**Review by Borda — APPROVED** [review]
mind add chlog :]
also pls keep the bullet-list form template

**Review by akihironitta — APPROVED** [review]

**Comment by Chizuchizu** [comment]
@Borda 

Thanks for the review.

I've added bulle-list and CHANGELOG.md.

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,6 +62,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed multiple early stopping callbacks ([#6197](https://github.com/PyTorchLightning/pytorch-lightning/pull/6197))
 
+- Fixed a bug regarding logger on TPU ([#6221](https://github.com/PyTorchLightning/pytorch-lightning/pull/6221))
```
Comment (akihironitta) [inline CHANGELOG.md:65]: How about making it a bit more specific?
```suggestion
- Fixed `AttributeError` when `logger=None` on TPU ([#6221](https://github.com/PyTorchLightning/pytorch-lightning/pull/6221))
```

**Review by akihironitta — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Chizuchizu) [reply CHANGELOG.md]: Thanks, good.

> How about making it a bit more specific?

Thanks, good.

**Review by SkafteNicki — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -62,6 +62,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 - Fixed multiple early stopping callbacks ([#6197](https://github.com/PyTorchLightning/pytorch-lightning/pull/6197))
 
+- Fixed `AttributeError` when `logger=None` on TPU ([#6221](https://github.com/PyTorchLightning/pytorch-lightning/pull/6221))
```
Comment (carmocca) [inline CHANGELOG.md:65]:
```suggestion

- Fixed `AttributeError` when `logger=None` on TPU ([#6221](https://github.com/PyTorchLightning/pytorch-lightning/pull/6221))
```

**Review by carmocca — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 3]
```diff
@@ -247,7 +247,7 @@ def __load_weights_on_main_process(self) -> None:
         self._model = model
 
     def _close_logger(self, trainer) -> None:
-        if hasattr(trainer, "logger"):
+        if hasattr(trainer, "logger.finalize"):
```
Comment (rohitgr7) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:250]: how about this?
```suggestion
        if trainer.logger is not None:
```

Comment (Chizuchizu) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:250]: If I set `logger=False`, won't AttributeError be raised as before?

Comment (rohitgr7) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:250]: If you set logger=False then trainer.logger is set to None.

Comment (Chizuchizu) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py]: Sorry, thanks

**Review by tchaton — APPROVED** [review]
Thanks for resolving my bug :)

**Review by s-rog — APPROVED** [review]

**Pull request merged**